compiler/optimizer/VPHandlers.cpp

```diff
@@ -6706,10 +6706,15 @@ TR::Node *constrainLdiv(OMR::ValuePropagation *vp, TR::Node *node)
       TR::VPConstraint *constraint = NULL;
       int64_t lhsConst = lhs->asLongConst()->getLong();
       int64_t rhsConst = rhs->asLongConst()->getLong();
-      if (lhsConst == TR::getMinSigned<TR::Int64>() && rhsConst == -1)
+      if (lhsConst == TR::getMinSigned<TR::Int64>() && rhsConst == -1 && !isUnsigned)
          constraint = TR::VPLongConst::create(vp, TR::getMinSigned<TR::Int64>());
       else if (rhsConst != 0L)
-         constraint = TR::VPLongConst::create(vp, TR::Compiler->arith.longDivideLong(lhsConst, rhsConst));
+         {
+         if (isUnsigned)
+            constraint = TR::VPLongConst::create(vp, ((uint64_t)lhsConst/(uint64_t)rhsConst));
```

Comment:
Why isn't this unsigned divide an operation in the Arith class like in the signed case?  e.g. `longUnsignedDivideLong` or some variant.

Reply:
As far as I can tell, the only purpose of the `ArithEnv` class is to allow OpenJ9 to override the behaviour of these operations by calling C helpers instead. Since OpenJ9 does not use unsigned opcodes, there's no reason for it to be overriding this behaviour.